Sources/ArcGISToolkit/Components/AR/FlyoverSceneView.swift

```diff
@@ -86,55 +74,104 @@ public struct FlyoverSceneView: View {
     }
 }
 
+/// An observable object that wraps an `ARSession` and provides the current frame.
+private class ObservableARSession: NSObject, ObservableObject, ARSessionDelegate {
+    /// The configuration used for the AR session.
+    private let configuration: ARWorldTrackingConfiguration
+    
+    /// The backing AR session.
+    private let session = ARSession()
+    
+    override init() {
+        configuration = ARWorldTrackingConfiguration()
+        configuration.worldAlignment = .gravityAndHeading
+        super.init()
+        session.delegate = self
+    }
+    
+    /// Starts the AR session.
+    func start() {
+        session.run(configuration)
+    }
+    
+    /// Pauses the AR session.
+    func pause() {
+        session.pause()
+    }
+    
+    /// The latest AR frame.
+    @Published
+    var currentFrame: ARFrame?
+    
+    func session(_ session: ARSession, didUpdate frame: ARFrame) {
+        currentFrame = frame
```

Comment:
The session captures video at a rate of 60 frames per second. Lowering the frame rate could improve the video quality by improving overall performance. The value `session.configuration?.videoFormat.framesPerSecond` is get-only. Maybe the value can be overridden or the [preferredFramesPerSecond](https://developer.apple.com/documentation/metalkit/mtkview/1536027-preferredframespersecond) could be specified on the metal view. When presenting the ARSCNView the frames can be set directly with SCNView [preferredFramesPerSecond](https://developer.apple.com/documentation/scenekit/scnview/1621205-preferredframespersecond). This change could be addressed in a separate PR.

Reply:
I agree, not for this PR. Also might not be something we'd want to change. We'd need to see.